internal/netpoll/epoll.go

```diff
@@ -37,11 +37,12 @@ import (
 
 // Poller represents a poller which is in charge of monitoring file-descriptors.
 type Poller struct {
-	fd             int    // epoll fd
-	wfd            int    // wake fd
-	wfdBuf         []byte // wfd buffer to read packet
-	netpollWakeSig int32
-	asyncTaskQueue queue.AsyncTaskQueue
+	fd                  int    // epoll fd
+	wfd                 int    // wake fd
+	wfdBuf              []byte // wfd buffer to read packet
+	netpollWakeSig      int32
+	asyncTaskQueue      queue.AsyncTaskQueue // queue with low priority
+	priorAsyncTaskQueue queue.AsyncTaskQueue // queue with high priority
```

Comment:
I would name them `dataQueue` and `taskQueue` or `asyncDataQueue` and `asyncTaskQueue`.

However there is a problem to use separate queues - what if a client does this in one call:
1. Sends some data
2. Close the connection

The expecation is to send the data then close the connection. However gnet would close the conn first as the task is in the priority queue.

Reply:
you've made a point here, I missed that scenario.

The main reason I introduce a new queue is to avoid starvation, there might be tons of `AsyncWrite` calls and few other tasks like `Close` or `Wake` which have higher priorities and we don't want them to be delayed too long by those tasks with lower priorities.

I reckon that I have to figure out a way to take care of both cases.